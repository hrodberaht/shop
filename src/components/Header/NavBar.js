import React, { Component } from 'react';
import { NavLink, Link } from 'react-router-dom';
import ConnectedAuthorization from '../Auth/Authorization';

export default class NavBar extends Component {
  render() {
    return (
      <div className="nav-bar">
        <ul className="nav-bar__list">
          <li className="item">
            <NavLink to="/">Shop</NavLink>
          </li>
          <li className="item">
            <NavLink to="/orders">Orders</NavLink>
          </li>
          <li className="item">
            <NavLink to="/cart">Cart</NavLink>
          </li>
          <ConnectedAuthorization
            render
            withRoleAdmin={(
              <li className="item">
                <Link to="/admin">Admin</Link>
              </li>
)}
          />
          <ConnectedAuthorization
            logout={logout => (
              <li className="item">
                <Link to="/" onClick={logout}>
                  Logout
                </Link>
              </li>
            )}
          />
        </ul>
      </div>
    );
  }
}
