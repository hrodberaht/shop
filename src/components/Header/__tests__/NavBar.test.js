import NavBar from '../NavBar';

const setup = buildSetup(NavBar, {});

describe('<NavBar />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
