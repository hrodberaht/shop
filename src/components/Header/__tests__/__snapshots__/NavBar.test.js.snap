// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render without crash 1`] = `
<div
  className="nav-bar"
>
  <ul
    className="nav-bar__list"
  >
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Shop
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/orders"
      >
        Orders
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/cart"
      >
        Cart
      </NavLink>
    </li>
    <Connect(Authorization)
      render={true}
      withRoleAdmin={
        <li
          className="item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/inventory"
          >
            Inventory
          </NavLink>
        </li>
      }
    />
    <Connect(Authorization)
      logout={[Function]}
    />
  </ul>
</div>
`;
